omc3/model/model_creators/lhc_model_creator.py

```diff
@@ -399,16 +400,32 @@ class LhcBestKnowledgeCreator(LhcModelCreator):  # -----------------------------
     jobfile: str = JOB_MODEL_MADX_BEST_KNOWLEDGE
     files_to_check: tuple[str] = (TWISS_BEST_KNOWLEDGE_DAT, TWISS_ELEMENTS_BEST_KNOWLEDGE_DAT)
 
-    def check_options(self, opt) -> bool:
+    def prepare_options(self, opt) -> bool:
         accel: Lhc = self.accel
-        if accel.list_b2_errors:
-            errors_dir = AFS_B2_ERRORS_ROOT / f"Beam{accel.beam}"
-            for d in errors_dir.iterdir():
-                if d.suffix==".errors" and d.name.startswith("MB2022"):
-                    print(d.stem)
-            return False
 
-        return super().check_options(opt)
+        # Check given B2 errors ---
+        exists = False
+        if accel.b2_errors is not None: 
+            # check if user gave absolute path (! with_suffix does not work, due to '.' in name)
+            exists = Path(f"{accel.b2_errors!s}.errors").is_file()
+
+        if not exists:
+            afs_dir = AFS_B2_ERRORS_ROOT / f"Beam{accel.beam}"
+            if not AFS_B2_ERRORS_ROOT.is_dir():
+                raise AcceleratorDefinitionError(
+                    f"Given b2 errors do not exist and neither does the AFS path: '{afs_dir!s}'."
+                )
+
+            accel.b2_errors = check_folder_choices(  # returns full path
+                afs_dir,
+                msg="No valid b2 errors given.",
+                selection=accel.b2_errors, 
+                list_choices=opt.list_choices,
+                predicate=get_check_by_regex_func(r"MB2022.+\.errors$"),
```

Comment:
I'm not sure I like to see a `2022` hardcoded in there. Surely one day there is an update and someone else has to dig and figure out we hardcoded this. Maybe at least some `20XX` kind of pattern?

Reply:
I know, but that's how it was before and I don't feel inclined to change this now. There are a bunch of other files in the folder that we do not want to use... A change for a later time

Reply:
Yeah I get that. Can you add a TODO comment just above this call?